src/Tizen.NUI.Components/Controls/NumberPicker.cs

```diff
@@ -0,0 +1,504 @@
+                Layout = new LinearLayout()
+                {
+                    LinearOrientation = LinearLayout.Orientation.Vertical,
+                },
+                //FIXME: Need to expand as many as possible;
+                //       When user want to start list middle of the list item. currently confused how to create list before render.
+                ScrollAvailableArea = new Vector2(0, 10000),
+                Name = "pickerScroller",
+            };
+            pickerScroller.Scrolling += OnScroll;
+            pickerScroller.ScrollAnimationEnded += OnScrollAnimationEnded;
+            pickerScroller.ScrollAnimationStarted += OnScrollAnimationStarted;
+
+            itemList = new List<TextLabel>();
+            
+            minValue = maxValue = currentValue = 0;
+            displayedValues = null;
+            //Those many flags for min, max, value method calling sequence dependendy.
+            needItemUpdate = true;
+            displayedValuesUpdate = false;
+            onAnimation = false;
+
+            Add(pickerScroller);
+            addLine();
+        }
+
+        /// <summary>
+        /// Dispose NumberPicker and all children on it.
+        /// </summary>
+        /// <param name="type">Dispose type.</param>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        protected override void Dispose(DisposeTypes type)
+        {
+            if (disposed)
+            {
+                return;
+            }
+
+            if (type == DisposeTypes.Explicit)
+            {
+                if (itemList != null)
+                {
+                    foreach (TextLabel textLabel in itemList)
+                    {
+                        Console.WriteLine("Item Del " + textLabel.Text);
+                        if (pickerScroller) pickerScroller.Remove(textLabel);
+                        Utility.Dispose(textLabel);
+                    }
+
+                    itemList = null;
+                }
+
+                if (pickerScroller != null)
+                {
+                    Remove(pickerScroller);
+                    Utility.Dispose(pickerScroller);
+                    pickerScroller = null;
+                }
+
+                Remove(upLine);
+                Utility.Dispose(upLine);
+                Remove(downLine);
+                Utility.Dispose(downLine);
+            }
+
+            base.Dispose(type);
+        }
+
+        /// <summary>
+        /// An event emitted when NumberPicker value changed, user can subscribe or unsubscribe to this event handler.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public event EventHandler<NumberPickerValueChangedEventArgs> NumberPickerValueChanged;
+
+        //TODO Fomatter here
+
+        /// <summary>
+        /// The values to be displayed instead of numbers.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public String[] DisplayedValues
```

Comment:
please change the interface as shown warning in line 178 : **Properties should not return arrays**

Reply:
To avoid this warning, i'm returning cloned array. but this warning still show up.

Reply:
Could you check this reference?

https://docs.microsoft.com/ko-kr/dotnet/fundamentals/code-analysis/quality-rules/ca1819
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections

Reply:
I thought it was a false alarm because it returned an unmodifiable array of strings. thanks the links